### DISCUSSION TITLE
prisma.$connect() with nodejs cluster

### DISCUSSION THREAD

**Discussion by vimalmistry** [prisma/prisma#9008] [Q&A]
I am getting this error while running my koa.js app as cluster.

```
Address already in use (os error 48)
/node_modules/@prisma/client/runtime/index.js:29704
                  const err = new import_PrismaClientInitializationError.PrismaClientInitializationError(json.message, this.clientVersion);
                              ^
```

Code
```js


const app = main();

if (cluster.isPrimary) {
   // Fork workers.
   for (let i = 0; i < os.cpus().length; i++) {
      cluster.fork();
   }

   cluster.on("exit", function (worker, _code, _signal) {
      console.log("worker " + worker.process.pid + " died");
   });
} else {
   app.listen(PORT, () => {
      console.log(`Server Running http://127.0.0.1:${PORT}`);
   });
}


```

I am using `prisma.$connect()` in main function.

Do I actually need ` `prisma.$connect()`` in code?

**Comment by janpio** [comment]
No, running the first query will also connect implicitly.

Does just using the code you posted above lead to the error? Can you provide the missing bits so we can fully reproduce please?

**Comment by vimalmistry** [comment]
@janpio 
Sorry. I forgot to mention prisma file code.

Here is code of prisma file.
```js
import { PrismaClient } from "@prisma/client";

const db = new PrismaClient({ log: ["query", "info", "warn", "error"] });
db.$use(async (params: any, next: any) => {
   const before = Date.now();
   const result = await next(params);
   const after = Date.now();
   console.log(
      `Query ${params.model}.${params.action} took ${after - before}ms`,
   );
   return result;
});
const { user: User, post: Post } = db;

export { db, User, Post };

```

And here is main function code. It is just a simple koa.js server.

```js
async function main() {
   const app = new Koa();
   await db.$connect();
   app.use(koaBody({ multipart: true }));
   app.use(mainRoutes.routes());
   return app;
}
```

**Reply by janpio** [reply]
Can you please open a new issue and include all that information there? Looks like something we should reproduce and see if there is something broken and to be fixed.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
